packages/apollo-gateway/src/__tests__/integration/multiple-key.test.ts

```diff
@@ -0,0 +1,321 @@
+import gql from 'graphql-tag';
```

Comment:
These tests are a great start, but I think we could use a few more to make sure we fully cover the multiple key dependencies:

1. A test with more keys than are needed by a query to fufill (i.e. three keys, but only two are needed to fetch the data)

2. A test where a service `provides` the additional key needed to fetch (i.e. if `actuaryService` provides the `id`)

3. A test with a compound key as one of the multiple keys

Reply:
Honestly pretty confused on general use of `@provides`... 

If my `actuaryService`'s SDL is:
```gql
    extend type User @key(fields: "ssn") {
      ssn: ID! @external
      risk: Float
    }
```
how do I even say it provides `id`?

I was thinking extending query with something like:
```
    extend type Query {
      mostSecure: User @provides(fields: "id")
    }
```
but that gives me a validation error on `Query.mostSecure -> A @provides selects id, but Query.id could not be found`. Which seems wrong, because I'm trying to select on User, not Query.